internal/dcache_lib/rpc/client/connection.go

```diff
@@ -0,0 +1,155 @@
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package client
+
+import (
+	"context"
+	"crypto/tls"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache_lib/rpc/gen-go/dcache/models"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache_lib/rpc/gen-go/dcache/service"
+	"github.com/apache/thrift/lib/go/thrift"
+)
+
+// Connection struct holds the Thrift connection to a node
+// This is used to make RPC calls to the node
+type Connection struct {
+	nodeID      string                      // Node ID of the node this connection is for, can be used for debug logs
+	nodeAddress string                      // Address of the node this connection is for
+	ctx         context.Context             // Context for the connection
+	Transport   thrift.TTransport           // Transport is the Thrift transport layer
+	Client      *service.ChunkServiceClient // Client is the Thrift client for the ChunkService
+}
+
+// NewConnection creates a new Thrift connection to a node
+func NewConnection(nodeID string, nodeAddress string) (*Connection, error) {
+	log.Debug("Connection::NewConnection: Creating new connection to node %s at %s", nodeID, nodeAddress)
+
+	protocolFactory := thrift.NewTBinaryProtocolFactoryConf(nil)
+	transportFactory := thrift.NewTTransportFactory()
+
+	var transport thrift.TTransport
+	cfg := &thrift.TConfiguration{
+		TLSConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	// if secure {
+	// 	transport = thrift.NewTSSLSocketConf(addr, cfg)
+	// }
+
+	transport = thrift.NewTSocketConf(nodeAddress, cfg)
+	transport, err := transportFactory.GetTransport(transport)
+	if err != nil {
+		log.Err("Connection::NewConnection: Failed to create transport [%v]", err.Error())
+		return nil, err
+	}
+
+	iprot := protocolFactory.GetProtocol(transport)
+	oprot := protocolFactory.GetProtocol(transport)
+	client := service.NewChunkServiceClient(thrift.NewTStandardClient(iprot, oprot))
+
+	conn := &Connection{
+		nodeID:      nodeID,
+		nodeAddress: nodeAddress,
+		ctx:         context.Background(), // TODO: check if context with cancel is needed
+		Transport:   transport,
+		Client:      client,
+	}
+
+	err = conn.Transport.Open()
+	if err != nil {
+		log.Err("Connection::NewConnection: Failed to open transport [%v]", err.Error())
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+// Close closes the Thrift connection to the node
+func (c *Connection) Close() error {
+	err := c.Transport.Close()
+	if err != nil {
+		log.Err("Connection::Close: Failed to close transport [%v]", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// IsAlive checks if the connection to the node is alive
+func (c *Connection) IsAlive(req *models.HelloRequest) bool {
```

Comment:
This package should only export RPC method which take request param struct and return a response struct.
Higher level calls like IsNodeAlive() could be implemented using the Hello RPC f.e., and typically it will be implements by some higher level package using the rpc_client package.

Reply:
Done